src/main/java/uk/gov/companieshouse/service/DissolutionCreator.java

```diff
@@ -38,4 +41,10 @@ public DissolutionCreateResponse create(DissolutionCreateRequest body, String co
 
         return responseMapper.mapToDissolutionCreateResponse(dissolution);
     }
+
+    public DissolutionGetResponse get(String companyNumber) {
+        Optional<Dissolution> dissolution = repository.findByCompanyNumber(companyNumber);
+
+        return responseMapper.mapToDissolutionGetResponse(dissolution.get());
```

Comment:
I would have this function return an Optional and let the controller react with a 404 if Optional.empty is returned

```
return repository
  .findByCompanyNumber(companyNumber)
  .map(dissolution -> responseMapper.mapToDissolutionGetResponse(dissolution))
```

Reply:
The controller already checks if the dissolution is present and throws 404. Would you like it to use this function instead?

Reply:
I would suggest you remove the existing check so that we can reduce the number of queries.